### ISSUE TITLE
Time Travel: Make PresentTime a real clock

### ISSUE THREAD

**Issue by realA10001986** [CircuitSetup/Time-Circuits-Display#15]
By using a "fake" year, it is pretty straight forward to use presentTime as a normal clock:

In tc_time.cpp, replace timeTravel with the following:

```
/* Time Travel: User enters a destination time.
 *  
 *  -) copy present time into departed time (where it freezes)
 *  -) copy destination time  to present time (where it continues to run as a clock)
 *
 *  This is called from tc_keypad.cpp 
 */

void timeTravel() {
    int tyr = 0;
    int tyroffs = 0;
        
    timetravelNow = millis();
    timeTraveled = true;
    beepOn = false;
    play_file("/timetravel.mp3", getVolume());
    allOff();

    // Copy present time to last time departed
    departedTime.setMonth(presentTime.getMonth());
    departedTime.setDay(presentTime.getDay());
    departedTime.setYear(presentTime.getYear());
    departedTime.setHour(presentTime.getHour());
    departedTime.setMinute(presentTime.getMinute());
    departedTime.save();

    // Copy destination time to present time
    // DateTime does not work before 2000 or after 2159
    // so we use yearOffs to fake a year with identical calendar within 2000-2159
    // RTC keeps running on 2000-2159, but display is skewed by yearOffs
    // upper limit 2150 to avoid overflow while clock running

    tyr = destinationTime.getYear();
    if (tyr < 2000) {
        while(tyr < 2000) {
            tyr += 28;
            tyroffs += 28;
        }
    } else if(tyr > 2150) {
        while(tyr > 2150) {
            tyr -= 28;
            tyroffs -= 28;
        }
    }
    
    if (!presentTime.isRTC()) presentTime.setRTC(true);
    
    presentTime.setYearOffset(tyroffs);
    
    rtc.adjust(DateTime(
            tyr,
            destinationTime.getMonth(),
            destinationTime.getDay(),
            destinationTime.getHour(),
            destinationTime.getMinute()
            )
    );

    Serial.println("timeTravel: Success, good luck!");
}
```

In clockdisplay.cpp:

```
void clockDisplay::setYear(uint16_t yearNum) {
    // Place LED pattern in year position in buffer, which is 4 and 5.
    _year = yearNum;
    yearNum -= _yearoffset;
    _displayBuffer[4] = makeNum(yearNum / 100);
    _displayBuffer[5] = makeNum(yearNum % 100);
}

void clockDisplay::setYearDirect(uint16_t yearNum) {
    // Place LED pattern in year position directly, which is 4 and 5.
    // useful for setting the year to the present time
    _year = yearNum;
    yearNum -= _yearoffset;
    directCol(4, makeNum(yearNum / 100));
    directCol(5, makeNum(yearNum % 100));
}

void clockDisplay::setYearOffset(int16_t yearOffs) {
    _yearoffset = yearOffs;
}
```
In clockdisplay.h add the marked lines:

```
    void setYear(uint16_t yearNum);
    void setYearOffset(int16_t yearOffs); // <----
    void setYearDirect(uint16_t yearNum);

[...]

    uint16_t _displayBuffer[8];  // Segments to make current time.

    uint16_t _year = 2021;     // keep track of these
    int16_t  _yearoffset = 0;  // Offset for faking years < 2000 or > 2159 <----
    
    uint8_t _month = 1;
```

I possibly omitted other definitions/declarations needed, but these are straight forward.

The RTC is loaded with a year with an identical calendar (as apparently happens every 28 years), and the year displayed is faked by _yearoffset.

**Comment by CircuitSetup** [comment]
Thanks for the tip! I'll test this soon!

**Comment by realA10001986** [comment]
I wrote you an email via the web contact form.

**Comment by realA10001986** [comment]
Made a fork with all of my changes, here:
https://github.com/realA10001986/Time-Circuits-Display-A10001986

**Comment by CircuitSetup** [comment]
This is fixed in PR #16
Thanks!
